src/tool_getparam.c

```diff
@@ -1610,15 +1611,15 @@ ParameterError getparameter(const char *flag, /* f or -long-flag */
         GetStr(&config->crlfile, nextarg);
         break;
       case 'k': /* TLS username */
-        if(curlinfo->features & CURL_VERSION_TLSAUTH_SRP)
-          GetStr(&config->tls_username, nextarg);
-        else
+        GetStr(&config->tls_username, nextarg);
+        cleanarg(nextarg);
+        if(!(curlinfo->features & CURL_VERSION_TLSAUTH_SRP))
```

Comment:
Shouldn't this be re-arranged to first check for feature support to avoid GetStr copying the sensitive string when not needed? cleanarg() should of course be invoked in both cases. Something like the below untested scribble:
```C
if(!(curlinfo->features & CURL_VERSION_TLSAUTH_SRP)) {
  cleanarg(nextarg);
  return PARAM_LIBCURL_DOESNT_SUPPORT;
}
GetStr(&config->tls_username, nextarg);
cleanarg(nextarg);
```

Reply:
Done.